### PR TITLE
Cgd 26 employee page

### PR DESCRIPTION
**Pull request by lnicepei**
Created employee page that displays editable data about an employee 

Deploy: https://cv-builder-ashen.vercel.app (open in new tab)

image

### CODE REVIEW

File: src/types/declarations.d.ts [hunk 1]
```diff
@@ -4,22 +4,61 @@ declare module '*.css' {
...
+  interface ProfilePageUser {
+    email?: string
+    profile: {
       id: string
-      name: string
-    } | null
+      first_name: string | null
+      last_name: string | null
+      avatar?: string
+      skills?:
```
Comment (Ascilator) [inline src/types/declarations.d.ts:42]: why not just [Skill]

File: src/types/declarations.d.ts [hunk 2]
```diff
@@ -4,22 +4,61 @@ declare module '*.css' {
...
+        | [
+            {
+              skill_name: string
+              mastery: string
+            }
+          ]
+        | []
+      languages?:
```
Comment (Ascilator) [inline src/types/declarations.d.ts:50]: same

File: src/types/declarations.d.ts [hunk 3]
```diff
@@ -4,22 +4,61 @@ declare module '*.css' {
...
+        | [
+            {
+              language_name: string
+              proficiency: string
+            }
+          ]
+        | []
+    }
+    cvs?: CV[] | []
```
Comment (Ascilator) [inline src/types/declarations.d.ts:59]: same

**Review by Ascilator — CHANGES_REQUESTED** [review]
изображение

Why i can edit someones profile? also why i don't see department etc

**Comment by lnicepei** [comment]
If you are logged in as uladzislau.yarmashuk@innowise-group.com(as admin), you can edit other profiles. If you log in as an employee, you can edit your own profile. And in case of a regular user, you cannot edit at all.

File: src/GraphQL/queries.ts [hunk 4]
```diff
@@ -33,3 +33,57 @@ export const EMPLOYEES_QUERY = gql`
...
+
+export const EMPLOYEE_QUERY = gql`
+  query GET_EMPLOYEE($id: ID!) {
+    user(id: $id) {
+      email
+
+      profile {
+        first_name
+        last_name
+
```
Comment (Ascilator) [inline src/GraphQL/queries.ts:45]: enter? why?

File: src/pages/EmployeePage/EmployeePage.tsx [hunk 5]
```diff
@@ -1,9 +1,39 @@
...
+    getEmployee(id, setEmployee)
+  }, [])
+
+  const tabItems = [
+    {
+      label: 'Profile',
+      key: 'profile',
+      children: <Profile id={id} employee={employee} setEmployee={setEmployee} canEdit={canEdit} />
+    },
+    { label: 'CVs', key: 'cvs', children: <CVs CVsData={employee?.cvs} canEdit={canEdit} /> }
```
Comment (Ascilator) [inline src/pages/EmployeePage/EmployeePage.tsx:33]: it should be the same style that in above

File: src/pages/EmployeePage/hooks/useGetEmployee.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { bindActionCreators } from 'redux'
+
+import { useLazyQuery } from '@apollo/client'
+
+import { EMPLOYEE_QUERY } from '@/GraphQL/queries'
+import { actionCreators, useAppDispatch } from '@/state'
+
+const useGetEmployee = () => {
```
Comment (Ascilator) [inline src/pages/EmployeePage/hooks/useGetEmployee.ts:8]: why this logic? why not just a hook that requests emp and returns him. Why do you need this complicate callback

**Review by Ascilator — CHANGES_REQUESTED** [review]

**Pull request merged**